src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -32,6 +32,23 @@ internal static class IntrinsicFunctions
         private static readonly Lazy<Regex> RegistrySdkRegex = new Lazy<Regex>(() => new Regex(@"^HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Microsoft SDKs\\Windows\\v(\d+\.\d+)$", RegexOptions.IgnoreCase));
 #endif // FEATURE_WIN32_REGISTRY
 
+        /// <summary>
+        /// Resolve the location of the NuGet.Frameworks assembly
+        /// </summary>
+        private static Assembly GetNuGetAssembly()
+        {
+            var assemblyDirectory = IsRunningFromVisualStudio() ?
+                Path.Combine(GetVsInstallRoot(), "Common7", "IDE", "CommonExtensions", "Microsoft", "NuGet") :
+                BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory;
+            return Assembly.LoadFile(Path.Combine(assemblyDirectory, "NuGet.Frameworks.dll"));
+        }
+
+        /// <summary>
+        ///  NuGet Types
+        /// </summary>
+        private static Type NuGetFramework = GetNuGetAssembly().GetType("NuGet.Frameworks.NuGetFramework");
+        private static Type NuGetFrameworkUtility = GetNuGetAssembly().GetType("NuGet.Frameworks.NuGetFrameworkUtility");
```

Comment:
Can you make these lazies instead of statics, so that we don't have to pay reflection cost on load for projects that don't use the new property functions?